src/sfloader/fluid_defsfont.c

```diff
@@ -1268,23 +1450,548 @@ fluid_zone_is_mod_identic(fluid_mod_t *mod, char *name)
+ *   First message indicates that m3 is invalid (because source src1 isn't linked
+ *   or mod is invalid or amount is 0).
+ *   When a path is without destination, all modulators from the beginning to the one
+ *   without destination are marked invalid (FLUID_PATH_VALID  = 0, amount = 0)
+ *   (e.g  m2,m6).
+ *
+ * - Circular path:
+ *   When a destination is a modulator already encountered this is a circular path
+ *   (e.g: CC-->m2-->m6-->m3-->m8-->m6). Two messages are displayed:
+ *      fluidsynth: warning: invalid circular path zone-name/mod6.
+ *      fluidsynth: warning: path without destination zone-name/mod2.
+ *   First message indicates that m6 is a modulator already encountered.
+ *   Second message indicates the modulator at the beginning of the path (e.g m2).
+ *   When a path is circular, all modulators from the beginning to the one
+ *   already encontered are marked invalid (FLUID_PATH_VALID  = 0, amount = 0)
+ *   (e.g  m2,m6,m3,m8).
+ *
+ * Other incomplete linked modulator paths are isolated.
+ * Isolated path begins with modulator mx having source src1 linked, with no
+ * others modulators connected to mx.
+ * These isolated modulator paths are still in list_mod but not in path table.
+ * They should be marked invalid later.
+ *
+ * The function searchs all linked path starting from the beginning of the path
+ * (ie. a modulator with source not linked) forward to the endind of the path
+ * (ie. a modulator connected to a generator).
+ * Search direction is the reverse that the one done in fluid_list_copy_linked_mod().
+ * The function is recursive and intended to be called the first time to
+ * start the search from the beginning of any path (see dest_idx, path_idx).
+ *
+ * @param list_name, list name used to prefix warning messages displayed.
+ *  if NULL, no message are displayed.
+ * @param list_mod, pointer on modulators list.
+ *  On output, invalid linked modulator are marked with amount value to 0.
+ * @param dest_idx, index of the destination linked modulator to search.
+ *  Must be - 1 at first call.
+ *   if < 0, search first modulator (i.e first linked modulateur).
+ *   if >= 0 index of the destination linked modulator to search.
+ * @param path, pointer on table for path registering.
+ *  On input, FLUID_PATH_CURRENT , FLUID_PATH_VALID must be initialized to 0.
+ *   FLUID_MOD_VALID to 1 indicates that the modulator is valid.
+ *  On output, path table contains bit for each modulator that indicates if
+ *   the modulator belongs to a linked path.
+ *  - no path (FLUID_PATH_CURRENT set to 0, FLUID_PATH_VALID set to 0) or
+ *  - valid complete paths (FLUID_PATH_CURRENT, FLUID_PATH_VALID set to 1) or
+ *  - invalid incomplete paths (FLUID_PATH_CURRENT set to 1, FLUID_PATH_VALID
+ *    set to 0).
+ *
+ * @return
+ *  - the number of linked modulators if any valid linked path exists.
+ *  - 0 if no linked path exists.
+*/
+static int
+fluid_check_linked_mod_path(char *list_name, fluid_mod_t *list_mod,
```

Comment:
Reminder for myself: Think of a better naming. `fluid_check_linked_mod_path` and `fluid_list_check_linked_mod` are too confusing.